### PR TITLE
#31 make SAML-Metadata (optionally) also available without Typo3-BE-Login

### PR DESCRIPTION
**Pull request by christophs78**

### CODE REVIEW

File: Classes/Middleware/SamlMiddleware.php [hunk 1]
```diff
@@ -65,9 +65,10 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
                 $loginType = 'BE';
             }
 
-            if (isset($GLOBALS['BE_USER']->user)) {
+            $extSettings = $this->settingsService->getSettings($loginType);
+
+            if (isset($GLOBALS['BE_USER']->user) || (isset($extSettings['metadataPublic']) && ($extSettings['metadataPublic'] === true))) {
```
Comment (georgringer) [inline Classes/Middleware/SamlMiddleware.php:70]: I suggest following code
```php
if (isset($GLOBALS['BE_USER']->user) || (bool)($extSettings['metadataPublic'] ?? false)) {
```

File: Configuration/TypoScript/setup.typoscript [hunk 2]
```diff
@@ -6,6 +6,9 @@ plugin.tx_mdsaml {
   settings {
     mdsamlSpBaseUrl =
 
+    // SAML-Metadata should be public available without Typo3-BE-Login
+    metadataPublic = false
```
Comment (georgringer) [inline Configuration/TypoScript/setup.typoscript:10]: I suggest as name `publicMetadata` but just a personal feeling

File: Classes/Middleware/SamlMiddleware.php (hunk 1 above)
Comment (christophs78) [reply Classes/Middleware/SamlMiddleware.php]: Thanks for your feedback!

**Comment by cdaecke** [comment]
Thanks four your pull request!

**Pull request merged**